src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs

```diff
@@ -141,7 +142,9 @@ private void PopulateAttributesFromStructuredEvent(CloudEvent cloudEvent, JObjec
                     continue;
                 }
 
-                var attribute = cloudEvent.GetAttribute(key);
+                // TODO: Validate that if the attribute is known, the token type is appropriate. Possibly
+                // only for known-to-spec-version attributes? If an integer extension attribute has a value of "1",
```

Comment:
Personally, I don't think that is necessary. If the CloudEvent originated from a different SDK, and you are registering an Integer extension, it feels like it should be enough for the value to be parsable into an int.

Reply:
Right. For extensions, I can see both sides - particularly given the way this SDK currently assumes "if we haven't seen something before, just store it as a string extension". There are cases where it *would* be nice to pick up issues early, but I think being lenient outweighs it.

For reserved attributes, as you've noted later, I think we should be stricter.

I won't change anything now, but I'll implement that in a follow-up PR and maybe note the deliberate decision for extension attributes.